BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginFormState.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.biometricloginsample
+
+sealed class LoginFormState
+
+class LoginState {
+    data class Failed(val usernameError: Int? = null,val passwordError: Int? = null)
+    data class Successful(val isDataValid: Boolean = false)
+}
```

Comment:
I'm not sure I understand this class. I would have expected the "state" to be either "SuccessfulLogin" or "FailedLogin" and then the failed would have `usernameError` and `passwordError` and Successful would have `isDataValid`?

Please add comments to all of these classes to explain what they're doing.

Reply:
It's not clear to me how what you are saying is different from what's there.

Reply:
I mean, it seems like `LoginState` shouldn't exist.

If the login succeeds, it should send back an instance of `SuccessfulLoginFormState`
If the login fails, it should send `FailedLoginFormState`

Reply:
so the sealed class should not exist?

Reply:
No, the opposite.

`LoginState` should not exist. It's not used anywhere. (Android Studio + Lint tell you this.) Please remove it. :)